Formula/yelp-tools.rb

```diff
@@ -15,30 +16,47 @@ class YelpTools < Formula
   end
 
   depends_on "gettext" => :build
-  depends_on "intltool" => :build
   depends_on "itstool" => :build
-  depends_on "libxml2" => :build
-  depends_on "libxslt" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
   depends_on "pkg-config" => :build
   depends_on "gtk+3"
+  depends_on "libxml2"
+  depends_on "python@3.9"
+
+  uses_from_macos "libxslt"
+
+  resource "lxml" do
+    url "https://files.pythonhosted.org/packages/e5/21/a2e4517e3d216f0051687eea3d3317557bde68736f038a3b105ac3809247/lxml-4.6.3.tar.gz"
+    sha256 "39b78571b3b30645ac77b95f7c69d1bffc4cf8c3b157c435a34da72e78c82468"
+  end
 
   resource "yelp-xsl" do
-    url "https://download.gnome.org/sources/yelp-xsl/3.38/yelp-xsl-3.38.1.tar.xz"
-    sha256 "b321563da6ab7fa8b989adaf1a91262059696316b4ddca2288fddcfed8dcdf67"
+    url "https://download.gnome.org/sources/yelp-xsl/40/yelp-xsl-40.0.tar.xz"
+    sha256 "361ecd4d33fccdb3bb08a687f60e5c3e909d2e9e3b022d844e049820d0cf62b0"
   end
 
   def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resource("lxml")
+    ENV.prepend_path "PATH", libexec/"bin"
+
     resource("yelp-xsl").stage do
-      system "./configure", "--disable-debug",
-                            "--disable-dependency-tracking",
+      system "./configure", "--disable-dependency-tracking",
                             "--disable-silent-rules",
                             "--prefix=#{prefix}"
       system "make", "install"
       ENV.append_path "PKG_CONFIG_PATH", "#{share}/pkgconfig"
     end
 
-    system "./configure", "--prefix=#{prefix}"
-    system "make", "install"
+    mkdir "build" do
+      system "meson", *std_meson_args, ".."
+      system "ninja", "-v"
+      system "ninja", "install", "-v"
+    end
+
+    # Replace shebang with virtualenv python
+    inreplace Dir[bin/"*"], "#!/usr/bin/python3", "#!#{libexec}/bin/python"
```

Comment:
Does this not work with our `rewrite_shebang` methods?

Reply:
I didn't try.
I had based the current logic on other formulae.

When I have the chance, I'll retry with `rewrite_shebang`

Reply:
Is there a way to use the virtualenv with `rewrite_shebang`?

Simply swapping out the line results in using Homebrew's Python, so it doesn't pick up the virtualenv packages.

Due to `meson`, I had trouble getting  `PYTHONPATH` manipulation working.

There doesn't seem to be any formula that combines `rewrite_shebang` + `meson` + Python package resources.